pygpu/gpuarray.pyx

```diff
@@ -1657,6 +1657,16 @@ cdef class GpuArray:
         """
         return pygpu_as_ndarray(self)
 
+    def __bool__(self):
+        if self.ga.nd == 0:
+            return True
+        if self.ga.nd == 1:
+            if self.ga.dimensions[0] == 0:
+                return False
+            if self.ga.dimensions[0] == 1:
+                return bool(numpy.asarray(self))
+        raise ValueError('The truth value of a multi-element array is ambiguous')
+
```

Comment:
Need tests.

Also, if nd==0, we should not always return True. We need to look at the data as in line 1667.

why not just do something like:

If self.ga.size == 0: return False
elif self.gz.size == 1: return bool(numpy.asarray(self))
raise ValueError(...)

Numpy accept do to bool on a matrix with only 1 element. I don't think it look at nd, but only at size.